### DISCUSSION TITLE
Request of firmware version releases for printers as the Ender 3 Max Neo and others

### DISCUSSION THREAD

**Discussion by teed3** [mriscoc/Ender3V2S1#862] [Ideas]
**Is your feature request related to a problem? Please describe.**
I have an Ender 3 v2 and it's working great with this firmware, but I'm thinking of buying an Ender 3 Max Neo, which has the important upgrades an Ev2 would want, plus a larger bed size, and it's on sale:
https://store.creality.com/products/ender-3-max-neo-3d-printer?spm=..page_2510923.header_1.1

I see it is running a 4.2.2 board and doesn't use a touch screen, so it should be fine. However, I see there are quite some settings to change to get a larger bed size. It would be nice if it had an easy support for the 300mm printers with a custom build or something.

It would also be nice if the website listed on the front page the set of Creality printers that should work out of the box. It's a little confusing the documentation only mentions two models. I'm new to this codebase (and printer) and it's comforting to see in print what is known to work.

_Originally posted by @KeithCu in https://github.com/mriscoc/Ender3V2S1/discussions/519_.

**Comment by ghost** [comment]
> I have an Ender 3 v2 and it's working great with this firmware, but I'm thinking of buying an Ender 3 Max Neo, 
You could also look for discussion about features (and support of some models).

> which has the important upgrades an Ev2 would want, plus a larger bed size,
Upgrade like changing the bed size would be achievable *without* custom compilation.

> I see it is running a 4.2.2 board and doesn't use a touch screen, so it should be fine. However, I see there are quite some settings to change to get a larger bed size. I

You could see this on your E 3 v2 under Advanced section Physical settings.
In Discussion, if it exists you could compare to your measured values like probe offset or mesh inset.

See this: https://youtu.be/fl5rRF7hlUg?t=521 vv screenshot from there
![image](https://user-images.githubusercontent.com/86535475/232062082-69134a29-36fc-4a4d-a688-408f8308f9a3.png)

**Comment by TheWebMachine** [comment]
I can confirm that you don't need a custom compile for the larger models. I have the Plus and, while it did initially take a little effort to determine the Min/Max X/Y/Z values, it was simple and a one-time thing. Once you have your values, you can simply create yourself a config file to be reloaded whenever you flash a newer version of the firmware. The config file can contain all the values specific to your printer that deviate from defaults coded in the firmware.

You can find my notes on config files and an example of mine here:
https://github.com/mriscoc/Ender3V2S1/discussions/796#discussioncomment-5445779

**Comment by KeithCu** [comment]
I agree it's not that heard to figure out the values, and create a config file. However, it's also easy for this very nice program to either include those config files, or have a simple UI built into the program that runs on first launch which avoids having every new user have to figure this same stuff out. Many users want to focus on making and improving prints, and not learning Marlin config file format.

**Comment by mriscoc** [comment]
> I agree it's not that heard to figure out the values, and create a config file. However, it's also easy for this very nice program to either include those config files, or have a simple UI built into the program that runs on first launch which avoids having every new user have to figure this same stuff out. Many users want to focus on making and improving prints, and not learning Marlin config file format.

It is of course quite possible to get a custom version with all settings enabled for your printer, there is a [Patreon subscription](https://www.patreon.com/mriscoc/membership) for users who want to focus on their printing and they do not want to learn how to configure the firmware themselves.

**Comment by TheWebMachine** [comment]
> I agree it's not that heard to figure out the values, and create a config file. However, it's also easy for this very nice program to either include those config files, 

That is something we would have to all collectively contribute to and could be made available in this repo, as @mriscoc doesn't have all of these printers to collect those values himself. And you can't always rely on Creality's posted specs (I was able to squeeze several extra mm on each axis beyond their stated conservative min/max axis values on my Plus, thereby increasing mesh level area). However, some things truly are preference, like Park position, etc.

I have already posted my Plus config file but we would need others to write proper and accurate configs for all the other models and post them so they could be collected together and placed in the repo proper.

> or have a simple UI built into the program that runs on first launch which avoids having every new user have to figure this same stuff out.

This would be easier said than done, especially on space-constrained printers that use F4 boards. First launch wizards of any kind are major space wasters in this case; as you are devoting limited flash space for a one-time piece of code that could otherwise be used for advanced features like Linear Advance and UBL.

**Comment by mriscoc** [comment]
Because it is not a firmware issue, I'm moving this thread to the Discussions section.

**Comment by KeithCu** [comment]
I'm not really sure how much space it takes to store a few config parameters. Anyway, I don't really care about the solution, I just think it should  be very easy to get this program going on an E3 Max printer without having to figure out the values.
